service/service.ml

```diff
@@ -45,8 +45,13 @@ module Make (Opam_repo : Opam_repository_intf.S) = struct
     let dispose (worker : Lwt_process.process) =
       let pid = worker#pid in
       Fmt.epr "Terminating worker %d@." pid;
-      worker#terminate;
-      worker#status >|= fun _ -> Fmt.epr "Worker %d finished@." pid
+      worker#close >|= function
```

Comment:
I'm not sure `close` terminates the worker, so you might need to use `worker#terminate` before ?

Reply:
Indeed!